src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.id

```diff
@@ -28,16 +31,50 @@ type StoragePowerActor struct {
 
     SuspendMiner(address addr.Address)
 
-    // UpdateStoragePowerSubsystem(Tipset, StateTree)
+    expectedConsensus ExpectedConsensus
+    ReportConsensusFault(
+        slasherAddr  addr.Address
+        faultType    ConsensusFaultType
+        proof        [block.Block]
+    )
+    ReportUncommittedPowerFault(cheaterAddr addr.Address, numSectors UVarint)
+    PowerTable
 
-    // VerifyElectionProof(base.Address, base.ElectionProof)	bool
+    // VerifyElectionProof(addr.Address, ElectionProof)	bool
 
     // TODO: old methods, should be removed
     // UpdateStoragePowerSubsystem(Tipset, StateTree)
-    // VerifyElectionProof(addr.Address, base.ElectionProof)	bool
+    // VerifyElectionProof(addr.Address, ElectionProof)	bool
     // NotifyStorageFault(addr.Address)
-    // TryPublishBlock(StateTree)					Block
+    // TryPublishBlock(StateTree)					block.Block
+    CommitPledgeCollateral(deals [deal.StorageDeal])
+    DecommitPledgeCollateral(deals [deal.StorageDeal])
+
+    pledgeCollateralBalances {addr.Address: UVarint}
+}
+
+type PowerTable struct {
+    // all power here is always verified
+    RegisterMiner(
+        addr        addr.Address
+        pk          filcrypto.PubKey
+        sectorSize  sector.SectorSize
+    )
+    miners {addr.Address: StorageMiner}
+    GetMinerPower(addr addr.Address) StoragePower
+    GetTotalPower() StoragePower
+    GetMinerPublicKey(addr addr.Address) filcrypto.PubKey
+    IncrementPower(addr addr.Address, numSectors UVarint)
+    SuspendPower(addr addr.Address, numSectors UVarint)
+    UnsuspendPower(addr addr.Address, numSectors UVarint)
+    RemovePower(addr addr.Address, numSectors UVarint)
+    RemoveAllPower(addr addr.Address, numSectors UVarint)
```

Comment:
i dont think we need all these methods, let's only have the ones we absolutely need (smaller potential problem/audit surface)